randovania/games/samus_returns/gui/dialog/cosmetic_patches_dialog.py

```diff
@@ -1,41 +1,111 @@
+
+    def _open_color_picker(self, color: tuple, propertyName: str) -> None:
+        picker_result = QtWidgets.QColorDialog.getColor(QtGui.QColor(*color))
+        if picker_result.isValid():
+            color_tuple = (picker_result.red(), picker_result.green(), picker_result.blue())
+            if propertyName == "laser_locked":
+                self._cosmetic_patches = dataclasses.replace(self._cosmetic_patches, laser_locked_color=color_tuple)
+            elif propertyName == "laser_unlocked":
+                self._cosmetic_patches = dataclasses.replace(self._cosmetic_patches, laser_unlocked_color=color_tuple)
+            elif propertyName == "grapple_laser_locked":
+                self._cosmetic_patches = dataclasses.replace(
+                    self._cosmetic_patches, grapple_laser_locked_color=color_tuple
+                )
+            elif propertyName == "grapple_laser_unlocked":
+                self._cosmetic_patches = dataclasses.replace(
+                    self._cosmetic_patches, grapple_laser_unlocked_color=color_tuple
+                )
+            elif propertyName == "energy_tank":
+                self._cosmetic_patches = dataclasses.replace(self._cosmetic_patches, energy_tank_color=color_tuple)
+            elif propertyName == "aeion_bar":
+                self._cosmetic_patches = dataclasses.replace(self._cosmetic_patches, aeion_bar_color=color_tuple)
+            elif propertyName == "ammo_hud":
+                self._cosmetic_patches = dataclasses.replace(self._cosmetic_patches, ammo_hud_color=color_tuple)
+            self._update_color_squares()
+
+    def _update_color_squares(self) -> None:
+        mapping = [
+            (self.custom_laser_locked_color_square, self._cosmetic_patches.laser_locked_color),
+            (self.custom_laser_unlocked_color_square, self._cosmetic_patches.laser_unlocked_color),
+            (self.custom_grapple_laser_locked_color_square, self._cosmetic_patches.grapple_laser_locked_color),
+            (self.custom_grapple_laser_unlocked_color_square, self._cosmetic_patches.grapple_laser_unlocked_color),
+            (self.custom_energy_tank_color_square, self._cosmetic_patches.energy_tank_color),
+            (self.custom_aeion_bar_color_square, self._cosmetic_patches.aeion_bar_color),
+            (self.custom_ammo_hud_color_square, self._cosmetic_patches.ammo_hud_color),
+        ]
+        for box, new_color in mapping:
+            style = "background-color: rgb({},{},{})".format(*new_color)
+            box.setStyleSheet(style)
 
     @property
     def cosmetic_patches(self) -> MSRCosmeticPatches:
         return self._cosmetic_patches
 
-    def reset(self):
+    def _on_check_update(self, check: QtWidgets.QCheckBox, field_name: str, _: None) -> None:
+        self.preferences = dataclasses.replace(
+            self.preferences, **{field_name: check.isChecked()}  # type: ignore[arg-type]
+        )
```

Comment:
Is this function even used? Can't see it on a quick glance.

Reply:
Yeah it's definitely not used from what I can tell either. Just extra copy paste stuff